km52/Kholin_Nikita/laboratory1.sql

```diff
@@ -1,2 +1,114 @@
 -- LABORATORY WORK 1
 -- BY Kholin_Nikita
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+видаляти дані з таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+create user kholin identified by kholin_password;
+grant connect to kholin;
+grant delete on kholin.people to kholin;
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина має лікарняну картку, що містить записи про історію хвороби.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+create table people
+(
+  person_id char(10) not null,
+  name char(30) not null
+);
+
+create table cards
+(
+  card_id char(10) not null,
+  person_id char(10) not null
+);
+
+create table card_records
+(
+  card_record_id char(10) not null,
+  card_id char(10) not null
+);
+
+ALTER TABLE people ADD CONSTRAINT PK_People PRIMARY KEY (person_id);
+ALTER TABLE cards ADD CONSTRAINT PK_Cards PRIMARY KEY (card_id);
+ALTER TABLE card_records ADD CONSTRAINT PK_Card_Records PRIMARY KEY (card_record_id);
+
+ALTER TABLE cards
+ADD CONSTRAINT FK_Cards_People FOREIGN KEY (person_id) REFERENCES people (person_id);
+ALTER TABLE people
+ADD CONSTRAINT FK_Card_Records_Cards FOREIGN KEY (card_id) REFERENCES cards(card_id);
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+grant resource to kholin
+
```

Comment:
should be "create", "select", "insert":
`grant select any table to kholin;
grant create any table to kholin;
grant insert any table to kholin ;`